ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/VulnReports/OnDemandReportsTab.tsx

```diff
@@ -1,27 +1,127 @@
 import React, { useCallback } from 'react';
-import { Card, CardBody, PageSection, Text } from '@patternfly/react-core';
+import {
+    Card,
+    CardBody,
+    PageSection,
+    Pagination,
+    Text,
+    Toolbar,
+    ToolbarContent,
+    ToolbarItem,
+    useInterval,
+} from '@patternfly/react-core';
 
 import { getTableUIState } from 'utils/getTableUIState';
+import { ensureBoolean, ensureStringArray } from 'utils/ensure';
+import { getRequestQueryStringForSearchFilter } from 'utils/searchUtils';
 import useRestQuery from 'hooks/useRestQuery';
+import useURLPagination from 'hooks/useURLPagination';
+import useURLSort from 'hooks/useURLSort';
+import useURLSearch from 'hooks/useURLSearch';
+import useURLStringUnion from 'hooks/useURLStringUnion';
 import { fetchOnDemandReportHistory } from 'services/ReportsService';
 import PageTitle from 'Components/PageTitle';
+import ReportJobStatusFilter, {
+    ensureReportJobStatuses,
+    ReportJobStatus,
+} from 'Components/ReportJob/ReportJobStatusFilter';
+import MyJobsFilter from 'Components/ReportJob/MyJobsFilter';
 import OnDemandReportsTable from './OnDemandReportsTable';
 
+const sortOptions = {
+    sortFields: ['Report Completed Time'],
+    defaultSortOption: { field: 'Report Completed Time', direction: 'desc' } as const,
+};
+
+function createQueryFromReportJobStatusFilters(jobStatusFilters: string[]) {
+    const query: Record<string, string[]> = {
+        'Report State': [],
+    };
+
+    const jobStatusQueryMappings: Record<string, { category: string; value: string }> = {
+        PREPARING: { category: 'Report State', value: 'PREPARING' },
+        WAITING: { category: 'Report State', value: 'WAITING' },
+        ERROR: { category: 'Report State', value: 'FAILURE' },
+        DOWNLOAD_GENERATED: {
+            category: 'Report State',
+            value: 'GENERATED',
+        },
+    };
+
+    const reportJobStatuses = ensureReportJobStatuses(jobStatusFilters);
+
+    reportJobStatuses.forEach((jobStatus) => {
+        const queryMapping = jobStatusQueryMappings[jobStatus];
+        if (queryMapping) {
+            const { category, value } = queryMapping;
+            query[category].push(value);
+        }
+    });
+
+    return query;
+}
+
 function OnDemandReportsTab() {
-    // @TODO: Pass query, pagination, sorting to API call
-    const fetchOnDemandReportsHistoryCallback = useCallback(() => fetchOnDemandReportHistory(), []);
-    const { data, isLoading, error } = useRestQuery(fetchOnDemandReportsHistoryCallback);
+    const { page, perPage, setPage, setPerPage } = useURLPagination(10);
+    const { sortOption, getSortParams } = useURLSort(sortOptions);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const [isViewingOnlyMyJobs, setIsViewingOnlyMyJobs] = useURLStringUnion('viewOnlyMyJobs', [
+        'false',
+        'true',
+    ]);
```

Comment:
Nit for future consideration - is it possible we would need to support more than two exclusive states here? Maybe something like "All_Jobs", "Only_My_Jobs", "My_Teams_Jobs", etc.? Hypothetical but it might be more extensible to use more descriptive strings in the union versus the boolean strings.